syntax = "proto3";

import "google/protobuf/empty.proto";

package myMessage;

service MyMessage {

rpc Put(KeyAndValue) returns (Empty){}

rpc Get(Key) returns (Value) {}

rpc notifyToManager(StorageInfo) returns (Empty) {}

/* // storage sends this
rpc RequestToManager(StorageInfo) returns (ManagerResponse){}

// storage sends this
rpc SendDataToManager(ValueWithVersion) returns (google.protobuf.Empty) {} */

}

message KeyAndValue {
  string key = 1;
  repeated string value = 2;
}

message Key{
  string key = 1;
}

message Value{
  repeated string value = 1;
}

message StorageInfo{
  string ip = 1;
}

message ManagerResponse{
  uint32 type = 1; // either PUT or GET
  string key = 2;
  repeated string value = 3;
  repeated string otherNodes = 4;
}

message ValueWithVersion{
  string clock = 1;
  repeated string value = 2;
}

message Empty{

}

/* message clientData {
  int32 ip = 1;
}

message serverResponse {
  string res = 1;
} */
